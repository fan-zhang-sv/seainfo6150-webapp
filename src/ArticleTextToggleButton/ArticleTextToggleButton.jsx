import React, {useState} from "react";
import styles from "./ArticleTextToggleButton.module.css";

const ArticleTextToggleButton = (props) => {
    return <button className={styles.uglyButton} onClick={props.onClick} type="button">{props.buttonText}</button>
}

export default ArticleTextToggleButton