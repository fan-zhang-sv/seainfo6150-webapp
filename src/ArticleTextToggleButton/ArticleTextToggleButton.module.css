.button {
  font-size: 16px;
  background: #c00;
  border-radius: 10px;
  color: #fff;
  padding: 10px;
}

/* Small devices (landscape phones, 576px and up) */
@media screen and (min-width: 576px) {
  .button {
    font-size: 14px;
  }
}

/* Medium devices (tablets, 768px and up) */
@media screen and (min-width: 768px) {
  .button {
    font-size: 13px;
  }
}

/* Large devices (desktops, 992px and up) */
@media screen and (min-width: 992px) {
  .button {
    font-size: 12px;
  }
}
