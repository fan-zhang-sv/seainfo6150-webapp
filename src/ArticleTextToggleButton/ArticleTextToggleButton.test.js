import React from "react";
import ArticleTextToggleButton from "./ArticleTextToggleButton";

describe("ArticleTextToggleButton tests", () => {

  it("renders correctly with text showing", () => {
    const { container } = render(<ArticleTextToggleButton
        buttonText={"Show less"}
        onClick={()=>{}}
      />);
    expect(container).toMatchSnapshot();
  });
  
});