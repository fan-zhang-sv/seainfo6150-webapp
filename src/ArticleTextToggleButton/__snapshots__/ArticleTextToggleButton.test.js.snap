// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTextToggleButton tests renders correctly with text showing 1`] = `
<div>
  <button
    class="button"
  >
    Show less
  </button>
</div>
`;
